src/config/mod.rs

```diff
@@ -72,17 +72,67 @@ impl<'a> From<&'a ConfigPath> for &'a PathBuf {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct ComponentId {
+    pub name: String,
+}
+
+impl ComponentId {
+    pub fn global<S: ToString>(name: S) -> Self {
+        Self {
+            name: name.to_string(),
+        }
+    }
+}
+
+impl std::fmt::Debug for ComponentId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.name)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WithInputs<Input, Inner> {
+    pub inputs: Vec<Input>,
+    #[serde(flatten)]
+    pub inner: Inner,
+}
```

Comment:
I'm a bit skeptical of this helper type since it seems to add a decent amount of indirection around something relatively simple. What if instead we allowed a `ComponentId` to be deserialized from a raw string? That might be just as convenient and quite a bit simpler.

Reply:
yeah, but when de-serializing a pipeline, the `ComponentId` wont be able to know that it's in the `pipeline_name.toml` and therefore set the good `scope` value. But I guess this can be done in a second step.

Reply:
I applied your solution, looks great 👍